src/app/tasks/task.service.ts

```diff
@@ -1,34 +1,36 @@
 import { Task, Option, BasicTask } from "./task.model";
-import { IdGeneratorService } from "./id-generator-service";
 import { Injectable } from "@angular/core";
 import { HttpClient } from "@angular/common/http";
 import { beApiRoutes } from "../routes/be-api.routes";
-import { BehaviorSubject } from 'rxjs';
+import { BehaviorSubject, Observable } from 'rxjs';
 
 @Injectable()
 export class TaskService {
     
-    private taskTypesSubject = new BehaviorSubject<Option[]>([]);
-    private taskStatusesSubject = new BehaviorSubject<Option[]>([]);
-    private tasksSubject = new BehaviorSubject<Task[]>([]);
+    private taskTypes : Option[] = [];
+    private taskStatuses : Option[] = [];
+    private tasks : Task[] = []
+    public basicTasksSubject = new BehaviorSubject<BasicTask[]>([]);
 
-    constructor(private idGeneratorService: IdGeneratorService, private httpClient: HttpClient) {
+    constructor(private httpClient: HttpClient) {
         this.fetchDataFromDB();
     }
 
     private fetchDataFromDB() {
         // Get task types, task statuses and tasks from DB
         this.httpClient.get<Option[]>(beApiRoutes.taskStatuses).subscribe(responseTaskStatuses => {
-            this.taskStatusesSubject.next(responseTaskStatuses);  
+            this.taskStatuses = responseTaskStatuses;  
```

Comment:
don't subscribe inside services, it's very hard to manage subscriptions inside services

Reply:
you're effectively making 3 separate calls to BE instead of just 1, would be worth it to make a new endpoint to fetch all necessary data at once